### PR TITLE
Button Press animation scale on Android

### PR DESCRIPTION
**Pull request by osdnk**
On android, in two places I decided to make animation-related components and touchable buttons independent because hitSlop corresponds with margins of components, and sometimes shadows might be outside of the hitslop area.

### CODE REVIEW

**Comment by skylarbarrera** [comment]
Big fan, other than the animation timing curve here's a few issues I saw from a quick run through. Most have to do with shadows, and element size's changing from previous.

shadows charts: https://cloud.skylarbarrera.com/Screenshot_20201125-125236_Rainbow.jpg

shadows receive: https://cloud.skylarbarrera.com/Screenshot_20201125-125523_Rainbow.jpg

shadows WC: https://cloud.skylarbarrera.com/Screenshot_20201125-125415_Rainbow.jpg

size issue swap: https://cloud.skylarbarrera.com/Screenshot_20201125-125059_Rainbow.jpg
text overflow due to sizing: https://cloud.skylarbarrera.com/Screenshot_20201125-125442_Rainbow.jpg

the import/paste button has a small pixel on the top left of the element, isnt showing on my screenshots for some reason but it is there on device.

I think you inadvertently fixed the All button issue(need to remove fix): https://cloud.skylarbarrera.com/Screenshot_20201125-125310_Rainbow.jpg

File: src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js [hunk 1]
```diff
@@ -1,75 +1,193 @@
...
+  children,
+  onPress,
+  onLongPress,
+  contentContainerStyle,
+  minLongPressDuration,
+  overflowMargin,
+  wrapperStyle,
+}) => {
+  const longPressTimer = useRef(false);
+  const isPressEventLegal = useRef(false);
```
Comment (brunobar79) [inline src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js:64]: #legalizePressEvents

File: src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js [hunk 2]
```diff
@@ -1,75 +1,193 @@
...
 
-const Wrapper = ({ children, radius, style }) =>
-  radius ? (
-    <RadiusWrapper borderRadius={radius} style={style}>
-      {children}
-    </RadiusWrapper>
-  ) : (
-    children
+  return (
+    <View style={[{ overflow: 'visible' }, wrapperStyle]}>
```
Comment (brunobar79) [inline src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js:131]: Is it worth moving the inline styles to use StyleSheet.create?

**Review by brunobar79 — APPROVED** [review]
🦃

**Comment by skylarbarrera** [comment]
supa close! got some margin issues mainly with the sheetActionButton
charts: https://cloud.skylarbarrera.com/Screenshot_20201126-115318_Rainbow.jpg

collectibles: https://cloud.skylarbarrera.com/Screenshot_20201126-115306_Rainbow.jpg

savings https://cloud.skylarbarrera.com/Screenshot_20201126-115252_Rainbow.jpg

WalletConnect: https://cloud.skylarbarrera.com/Screenshot_20201126-121124_Rainbow.jpg
^ White buttons still need some elevation/shadow 

Edit Wallet button animated weirdly: https://cloud.skylarbarrera.com/Screen_Recording_20201126-115353.mp4

Gas Selection got messed up: https://cloud.skylarbarrera.com/Screen_Recording_20201126-115235.mp4

all button still needs to be refixed on sendSheet(I can handle if you want to focus on the rest): https://cloud.skylarbarrera.com/Screenshot_20201126-115634_Rainbow.jpg

File: src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js (hunk 1 above)
Comment (osdnk) [reply src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js:64]: #PressEventsMatter

**Comment by osdnk** [comment]
@skylarbarrera how do you find it now?

**Comment by skylarbarrera** [comment]
Wallet Connect Margins: https://cloud.skylarbarrera.com/Screenshot_20201127-111544_Rainbow.jpg
Gas Placeholder: https://cloud.skylarbarrera.com/Screenshot_20201127-111216_Rainbow.jpg

File: src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js [hunk 3]
```diff
@@ -1,75 +1,197 @@
...
+      }, minLongPressDuration);
+    }
+
+    isPressEventLegal.current = true;
+  };
+  const handlePress = () => {
+    clearTimeout(longPressTimer.current);
+    if (longPressTimer.current) {
+      onPress();
+      console.log('!!!');
```
Comment (skylarbarrera) [inline src/components/animations/ButtonPressAnimation/ButtonPressAnimation.android.js:101]: 😘

**Comment by christianbaroni** [comment]
@osdnk is it possible this is making the whole app a bit slower? seeing more frame drops and delays than before, mainly when opening sheets/navigating

**Comment by osdnk** [comment]
Hm, it is possible, @christianbaroni.
Can you confirm you definitely find app slower w/o those changes?

**Review by skylarbarrera — APPROVED** [review]
Pressed every button in the app and this feels 💯 .  Once the following are fixed this is good 2 merge

Cloud button: blur overflow & Manual Back up button: 1 row: https://cloud.skylarbarrera.com/Screenshot_20201201-232437_Rainbow.jpg

I've saved these words button: https://cloud.skylarbarrera.com/Screenshot_20201201-233908_Rainbow.jpg

WC transaction buttons, need spacing: https://cloud.skylarbarrera.com/Screenshot_20201201-230446_Rainbow-1-.jpg

**Pull request merged**